### PR TITLE
Add GitHub actions for automated NPM release

### PR DESCRIPTION
**Pull request by thibmeu**
When a tagged commit is pushed to main, an NPM release should be triggered.

`node.yml` has been moved to `tests.yaml` to be more descriptive.

### CODE REVIEW

**Comment by thibmeu** [comment]
the last commit adds instruction to generate a new NPM_TOKEN

**Review by lukevalenta — APPROVED** [review]

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+name: Release
+
+on:
```
Comment (armfazh) [inline .github/workflows/release.yml:3]: here the trigger could be `on: release`
see https://docs.github.com/en/actions/publishing-packages/publishing-nodejs-packages#publishing-packages-to-the-npm-registry

**Review by armfazh — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yml (hunk 1 above)
Comment (thibmeu) [reply .github/workflows/release.yml:3]: `on: release` implies we create releases on GitHub, which I'm not sure we want to do.
Solely relying on tag and branch seems simpler and does not require interacting with GitHub.

Comment (armfazh) [reply .github/workflows/release.yml:3]: github helps with tagging, I have been used Github Releases for other projects, and it doesn't add too much complexity.

Comment (thibmeu) [reply .github/workflows/release.yml:3]: I've updated the release condition to be `on: release: types: [published]`

**Pull request merged**